Bottom: c4906b969c5b4cb5a5581ea26ac771988a8df770
Top:    fd0012f2d2114978488547b6e7c14d49c1ebb804
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 17:44:14 +0100

hid/gtk: Avoid the deprecated gdk_drawable_get_display()

Grab the active display using gdk_window_get_display() on the window
belonging to the drawing area.


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 89d50be..3e253cb 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -1542,9 +1542,8 @@ Benchmark (int argc, char **argv, Coord x, Coord y)
 {
   int i = 0;
   time_t start, end;
-  GdkDisplay *display;
-
-  display = gdk_drawable_get_display (gport->drawable);
+  GdkWindow *window = gtk_widget_get_window (gport->drawing_area);
+  GdkDisplay *display = gdk_window_get_display (window);
 
   gdk_display_sync (display);
   time (&start);
